packages/block-editor/src/hooks/border-style.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import BorderStyleControl from '../components/border-style-control';
+import useEditorFeature from '../components/use-editor-feature';
+import { BORDER_SUPPORT_KEY } from './border';
+import { cleanEmptyObject } from './utils';
+
+/**
+ * Inspector control for configuring border style property.
+ *
+ * @param  {Object} props  Block properties.
+ * @return {WPElement}     Border style edit element.
+ */
+export const BorderStyleEdit = ( props ) => {
+	const {
+		attributes: { style },
+		setAttributes,
+	} = props;
+
+	if ( useIsBorderStyleDisabled( props ) ) {
+		return null;
+	}
+
+	const onChange = ( newBorderStyle ) => {
+		const newStyleAttributes = {
+			...style,
+			border: {
+				...style?.border,
+				style: newBorderStyle,
+			},
+		};
+
+		setAttributes( { style: cleanEmptyObject( newStyleAttributes ) } );
+	};
+
+	return (
+		<BorderStyleControl
+			value={ style?.border?.style }
+			onChange={ onChange }
+		/>
+	);
+};
+
+/**
+ * Determines if there is border style support.
+ *
+ * @param  {string|Object} blockType Block name or block type object.
+ * @return {boolean}                 Whether there is support.
+ */
+export const hasBorderStyleSupport = ( blockType ) => {
```

Comment:
this function is doubled

Reply:
I've refactored all the individual feature checks to use a single more generic function exported from `border.js`